### ISSUE TITLE
[Bug] 大量数据的折线图，前后不连续的点没有绘制

### ISSUE THREAD

**Issue by FanLu1994** [apache/echarts#18970]
### Version

5.3.3

### Link to Minimal Reproduction

https://echarts.apache.org/examples/zh/editor.html?code=PYBwLglsB2AEC8sDeAoWsAeBBDEDOAXMmurGAJ4gCmRA5AMYCGYVA5sAE7m0A0J6AE2aMiAbVoBZGL1i0AKgFcqM2gHUqAlXIAWClQDEOEFQGVmphdFoBdEgF8-6cjnxFUpMpRqyAbowA2SrT2jrB4VEZUhLCi_MQeHnjawADuJuQAtgBGwP4EAGYB4aEeQmAiMQCMAKwADDywDQBMTQAsDQ2VAMzVna0A7M0AbLXWJaQU1HT-ENDKcXYktnYA3EA

### Steps to Reproduce

链接里面即是复现

### Current Behavior

1. 折线图
2. 大量数据，几万，不能sample
3. 不能showSymbol，会非常卡顿
4. 如果一个点 前后都为空，那么这个点也不会被绘制

### Expected Behavior

1. 希望这个点能被绘制出来

### Environment

```markdown
- OS: window
- Browser: edge
- Framework: vue
```

**Comment by FanLu1994** [comment]
if i set showSymbol=true， my page will be very laggy。
i found a solution here： https://github.com/apache/echarts/pull/18815
but it cannot work for me,  because my xAxis type is time 😢

**Comment by helgasoft** [comment]
Good reference to #18815. PR has been updated and works with _type:'time'_. 
 [Demo Code](https://echarts.apache.org/examples/en/editor.html?version=PR-18815%40bacb997&code=PYBwLglsB2AEC8sDeAoWsAmBDMWDOApmAFzJrqx7ACuATgMYGkDa5FszA5AEwAMvATgD0ANiEBGbrG7F-nADSwALAGYAdCoC68thS59BoiVJWzeC2NGoAba9t3p9_YWMmwArGYvdxADnvsjjzORm4iXopWtgGBToauUgDsEdIivDHscS7GsL4p4mkZesHxOQL5vuk6gRwl2W7ivClRdtWxdaFS4uIpBpq6_egAvm0AHgCCoxB4pKjsYACeIEywnJAAtgSc5CPkC5PTs7C76IS0EAQzHLpzgXgAFsAA7gDKC-sARsDWpABmWNZCG12PQYNACPQwAA5GyAv4AoEOWCLZakTjWCDghRIvDvL7WF4QABeK3E7kUuM-3zRGAgWHWMAw2JqGPBL0W1hWSFgTwgGDA91I7mOwIoBGgoIwXKR6FGpF4ovYC1I4iRQ106vQ_SGAG4gA).
So what I said - _"There is no option for "lines without points" mixed with single points."_ - no longer applies, which is good news.  
🤞 for PR to be approved (soon). Praise to @linghaoSu for pulling that off 🏆
![image](https://github.com/apache/echarts/assets/13038071/93d2d7f2-76c1-49c0-9e73-06f8899a6462)

**Comment by linghaoSu** [comment]
> Good reference to #18815. PR has been updated and works with _type:'time'_. [Demo Code](https://echarts.apache.org/examples/en/editor.html?version=PR-18815%40bacb997&code=PYBwLglsB2AEC8sDeAoWsAmBDMWDOApmAFzJrqx7ACuATgMYGkDa5FszA5AEwAMvATgD0ANiEBGbrG7F-nADSwALAGYAdCoC68thS59BoiVJWzeC2NGoAba9t3p9_YWMmwArGYvdxADnvsjjzORm4iXopWtgGBToauUgDsEdIivDHscS7GsL4p4mkZesHxOQL5vuk6gRwl2W7ivClRdtWxdaFS4uIpBpq6_egAvm0AHgCCoxB4pKjsYACeIEywnJAAtgSc5CPkC5PTs7C76IS0EAQzHLpzgXgAFsAA7gDKC-sARsDWpABmWNZCG12PQYNACPQwAA5GyAv4AoEOWCLZakTjWCDghRIvDvL7WF4QABeK3E7kUuM-3zRGAgWHWMAw2JqGPBL0W1hWSFgTwgGDA91I7mOwIoBGgoIwXKR6FGpF4ovYC1I4iRQ106vQ_SGAG4gA). So what I said - _"There is no option for "lines without points" mixed with single points."_ - no longer applies, which is good news. 🤞 for PR to be approved (soon). Praise to @linghaoSu for pulling that off 🏆 ![image](https://user-images.githubusercontent.com/13038071/259840940-93d2d7f2-76c1-49c0-9e73-06f8899a6462.png)

You are right, modifiy directly will cause breaking change, so I add a new option `showDisconnectSymbol`, accepting `'auto' | true`, default is `'auto'`.

-  `'auto'` will follow the `showSymbol` behavior. 
- `true` will show the disconnect symbol although `showSymbol` is `false`
